RULES/core/context.go

```diff
@@ -105,6 +105,23 @@ type coverageReportInterface interface {
 	Build(ctx Context)
 }
 
+type TranslationUnit struct {
+	Source Path
+	Object OutPath
+	Flags []string
+}
+
+// AnalyzeInterface is an interface for targets cpmpatible with static analisys
+type AnalyzeInterface interface {
+	Units(ctx Context) []TranslationUnit
+	EnumerateDeps(ctx Context) []AnalyzeInterface
```

Comment:
why `EnumerateDeps` instead of simply `Deps`?

Reply:
I guess most test targets already have a `Deps` field that take an  array of `cc.Dep` so it would conflict with it. But it could probably be renamed to something related to static analysis  like `AnalysisDeps`.

Reply:
Similarly, I'd rename Units to TranslationUnits, since Units is just too generic.